drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/migration/mci_mapping_importer.rb

```diff
@@ -0,0 +1,106 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module HmisExternalApis::AcHmis::Importers::Migration
+  class MciMappingImporter
+    attr_accessor :io, :new_record_count
+
+    def initialize(io:)
+      self.io = io
+      self.new_record_count = 0
+    end
+
+    def run!
+      make_lookup_table!
+
+      mci_unique_ids.find_each do |mci_unique_id|
+        mapped = get_matching_mci_id(client_id: mci_unique_id.source_id, mci_id: lookup[mci_unique_id.value])
+
+        log_what_happened(client_id: mci_unique_id.source_id, mapped: mapped)
+
+        mapped.remote_credential ||= remote_credential
+
+        self.new_record_count += 1 if mapped.new_record?
+
+        mapped.save!
+      end
+
+      Rails.logger.info "#{self.new_record_count} records were added."
+    end
+
+    private
+
+    def make_lookup_table!
+      lookup
+    end
+
+    def mci_unique_ids
+      result = HmisExternalApis::ExternalId
+        .for_clients
+        .where(namespace: 'ac_hmis_mci_unique_id')
```

Comment:
I think it would be better to use the system id constants rather than having strings everywhere?
```suggestion
        .where(namespace: HmisExternalApis::AcHmis::WarehouseChangesJob::NAMESPACE)
```